packages/devextreme/testing/tests/DevExpress.ui.widgets/multiView.tests.js

```diff
@@ -1092,6 +1222,104 @@ QUnit.module('keyboard navigation', {
+        const $lastItem = $multiView.find(`.${MULTIVIEW_ITEM_CLASS}`).eq(2);
+
+        $multiView.focusin();
+        keyboardMock($multiView).keyDown('right');
+
+        assert.strictEqual(instance.option('selectedIndex'), 2, 'correct index is selected');
+        assert.roughEqual(position($multiView), position($lastItem), 1, 'container moved to proper position');
+    });
+
+    QUnit.test('selectedIndex should be correct when one item is not visible using left arrow key', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [
+                { text: '1', visible: true },
+                { text: '2', visible: false },
+                { text: '3', visible: true }
+            ],
+            selectedIndex: 2,
+            focusStateEnabled: true
+        });
+        const instance = $multiView.dxMultiView('instance');
+        const $firstItem = $multiView.find(`.${MULTIVIEW_ITEM_CLASS}`).eq(0);
+
+        $multiView.focusin();
+        keyboardMock($multiView).keyDown('left');
+
+        assert.strictEqual(instance.option('selectedIndex'), 0, 'correct index is selected');
+        assert.roughEqual(position($multiView), -position($firstItem), 1, 'container moved to proper position');
+    });
+
+    QUnit.test('when only one item is visible, does not move the current visible item using all supported kbn keys', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [
+                { text: '1', visible: true },
+                { text: '2', visible: false }
+            ],
+            loop: true,
+            focusStateEnabled: true
+        });
+        const instance = $multiView.dxMultiView('instance');
+        const $itemContainer = $multiView.find(toSelector(MULTIVIEW_ITEM_CONTAINER_CLASS));
+
+        $multiView.focusin();
+        keyboardMock($multiView).keyDown('left');
+
+        assert.strictEqual(position($itemContainer), 0, 'container did not move');
+        assert.strictEqual(instance.option('selectedIndex'), 0, 'selectedIndex is not changed');
+
+        keyboardMock($multiView).keyDown('right');
+
+        assert.strictEqual(position($itemContainer), 0, 'container did not move');
+        assert.strictEqual(instance.option('selectedIndex'), 0, 'selectedIndex is not changed');
+
+        keyboardMock($multiView).keyDown('home');
+
+        assert.strictEqual(position($itemContainer), 0, 'container did not move');
+        assert.strictEqual(instance.option('selectedIndex'), 0, 'selectedIndex is not changed');
+
+        keyboardMock($multiView).keyDown('end');
+
+        assert.strictEqual(position($itemContainer), 0, 'container did not move');
+        assert.strictEqual(instance.option('selectedIndex'), 0, 'selectedIndex is not changed');
+    });
+
+    QUnit.test('item switching should go to first and last visible element when clicking home and end button correspondingly', function(assert) {
```

Comment:
[evergreen test]
If u break code related to `visible` property processing this test will be still green because first and last item are visible.
You should hide first item for testing `home` and last item for testing `end` (supposed to be 2 different tests)

Reply:
[84aed14](https://github.com/DevExpress/DevExtreme/pull/28137/commits/84aed14aa4791d6603352b53a1a6cc463c2f2d28)